src/main/java/com/Elessar/app/server/MyServer.java

```diff
@@ -38,7 +38,7 @@ public MyServer(String serverName, int port, MyDatabase db, LoadingCache<String,
         this.port = port;
         this.db = db;
         this.httpClient = new HttpClient(new NetHttpTransport().createRequestFactory());
-        this.msgSender = new DirectMsgSender(httpClient);
+        this.msgSender = new KafkaMsgSender(metricManager);
```

Comment:
Same as what I mentioned for ClientMain, here we should pass in a flag indicating pull vs. push and create sender based on the flag.